synapseclient/client.py

```diff
@@ -3468,6 +3516,34 @@ def setProvenance(
 
         Returns:
             An updated [synapseclient.activity.Activity][] object
+
+        Example: Using this function (DEPRECATED)
+            Setting provenance for an entity
+
+                activity = Activity(name="Analysis", description="Data processing")
+                updated_activity = syn.setProvenance("syn123", activity)
+
+        Example: Migration to new method
+            &nbsp;
+
+            ```python
+            from synapseclient import Synapse
+            from synapseclient.models import Activity
+
+            # Login to Synapse
+            syn = Synapse()
+            syn.login()
+
+            # Create an activity
+            activity = Activity(
+                name="Analysis",
+                description="Data processing"
+            )
+
+            # Store the activity and associate it with the entity
+            stored_activity = activity.store(parent="syn123")
```

Comment:
From a product perspective, should this be an "entity" function? I'm taking a look at the API and I'm wondering if this should be 

`entity.store_activity(Activity)` instead of `Activity.store(parent="syn2222")`

My understanding is that you can create activity objects without actually storing it at all to an entity (not that that's particularly helpful.

I believe this can be a separate ticket

Reply:
It is already available within the entities. See the `activity` field on https://python-docs.synapse.org/en/stable/reference/experimental/sync/file/?h=activity#synapseclient.models.File . What would occur is that you'd set your Activity object on the entity, and then store the entity. No specific method required like `store_activity`. Using this `activity.store` was a way to provide a 1-to-1 mapping for functionality in the event there are folks that have code that work this way instead.